trailblazer/services/tower/tower_api_service.py

```diff
@@ -1,148 +1,26 @@
-"""Module for Tower Open API."""
-
-import logging
-from pathlib import Path
-
-from trailblazer.clients.tower.models import (
-    TowerProcess,
-    TowerTask,
-    TowerTaskResponse,
-    TowerWorkflowResponse,
-)
-from trailblazer.clients.tower.tower_client import TowerApiClient
-from trailblazer.constants import TOWER_WORKFLOW_STATUS, FileFormat, TrailblazerStatus
-from trailblazer.exc import TowerRequirementsError
-from trailblazer.io.controller import ReadFile
-
-
-LOG = logging.getLogger(__name__)
+from trailblazer.clients.tower.models import TowerTasksResponse
+from trailblazer.clients.tower.tower_client import TowerAPIClient
+from trailblazer.services.tower.utils import create_job_from_tower_task, get_tower_workflow_id
+from trailblazer.store.models import Analysis, Job
+from trailblazer.store.store import Store
 
 
 class TowerAPIService:
     """Class communicating with NF tower regarding a given analysis (workflow)."""
 
-    def __init__(self, workflow_id: str, dry_run: bool = False):
-        self.workflow_id: str = workflow_id
-        self.dry_run: bool = dry_run
-        self._tower_client = None
-        self._response = None
-        self._tasks_response = None
-
-    @property
-    def tower_client(self) -> TowerApiClient:
-        """Returns a NF Tower client."""
-        if not self._tower_client:
-            self._tower_client = TowerApiClient(workflow_id=self.workflow_id)
-        return self._tower_client
-
-    @property
-    def response(self) -> TowerWorkflowResponse:
-        """Returns a workflow response containing general information about an analysis."""
-        if not self._response:
-            self._response = self.tower_client.workflow
-        return self._response
-
-    @property
-    def tasks_response(self) -> TowerTaskResponse:
-        """Returns a tasks response containing information about jobs submitted as part of an analysis.
-        A task response does not include all jobs that will be run as part of an analysis.
-        It only includes jobs that have been submitted a at given time that could be either
-        pending, running, completed or failed."""
-        if not self._tasks_response:
-            self._tasks_response = self.tower_client.tasks
-        return self._tasks_response
-
-    @property
-    def status(self) -> str:
-        """Returns the status of an analysis (also called workflow in NF Tower)."""
-        status: str = TOWER_WORKFLOW_STATUS.get(
-            self.response.workflow.status, TrailblazerStatus.ERROR
```

Comment:
The overall status for tower analyses seems to be mirrored by the workflow status, no its constituent jobs.